third_party/amd/lib/TritonAMDGPUTransforms/OptimizeEpilogue.cpp

```diff
@@ -38,11 +38,39 @@ using namespace mlir;
 
 namespace {
 
+bool isOneOperandElementwiseOp(Operation *op) {
+  if (llvm::isa<arith::ExtFOp, arith::ExtSIOp, arith::ExtUIOp, arith::FPToSIOp,
+                arith::FPToUIOp, arith::NegFOp, arith::SIToFPOp,
+                arith::TruncFOp, arith::TruncIOp, arith::UIToFPOp>(op))
+    return true;
+  if (llvm::isa<math::AbsFOp, math::AbsIOp, math::AtanOp, math::Atan2Op,
+                math::CeilOp, math::CosOp, math::SinOp,
+                math::CountLeadingZerosOp, math::CountTrailingZerosOp,
+                math::CtPopOp, math::ErfOp, math::ExpOp, math::Exp2Op,
+                math::ExpM1Op, math::FloorOp, math::LogOp, math::Log10Op,
+                math::Log1pOp, math::Log2Op, math::RsqrtOp, math::SqrtOp,
+                math::TanhOp>(op))
+    return true;
+  if (llvm::isa<triton::IntToPtrOp, triton::PtrToIntOp, triton::BitcastOp,
+                triton::FpToFpOp>(op))
+    return true;
+  if (auto externElementwiseOp = dyn_cast<triton::ExternElementwiseOp>(op))
+    return op->getNumOperands() == 1 &&
+           op->getNumResults() == 1 & externElementwiseOp.getPure();
+  return false;
+}
+
 // convert(val) : mma -> blocked
+// elementWiseOp(val) : blocked
+// ...
+// elementWiseOp(val) : blocked
 // tt.store(ptr, val, mask, ...) : blocked
 // ==>
 // convert(ptr) : blocked -> mma
 // convert(mask) : blocked -> mma
+// elementWiseOp(val) : mma
```

Comment:
same here

Reply:
I think it is better to name it `xmma` or something.

This pass check for new `triton::gpu::MmaEncodingTrait`, which is currently a part of every *MMA layout (both Nvidia and AMD)

Reply:
I think you mentioned nv path has removed this pass. So here we are looking for mfma specifically.

btw, is it still legal to check for triton::gpu::MfmaEncodingAttr?

Reply:
> btw, is it still legal to check for triton::gpu::MfmaEncodingAttr?

yes, but I think @joviliast will use this pass for WMMA as well, he will probably face same LDS issues as we had with MI hardware. RDNA3 LDS is still 64k

Reply:
Make sense

Reply:
Let me change this comment in next PR, I think it is a minor issue at this point

Reply:
here is the change: https://github.com/openai/triton/pull/3271/files#diff-410000ee7970cd30de554715f808fdda618099fc614b7a5ca778de7d0850d2e4L62